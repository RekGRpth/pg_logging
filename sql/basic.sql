create schema logging;
create extension pg_logging schema logging;

set pg_logging.buffer_position = 0;
set pg_logging.buffer_size = 2000;
set pg_logging.buffer_size = 1;
show pg_logging.buffer_position;

select logging.flush_log();
select * from logging.pg_log;

select 1/0;
show pg_logging.buffer_position;
select 'aaaaa'::int;
show pg_logging.buffer_position;
select * from logging.get_log(true);

select repeat('aaaaaaaaa', 20)::int;
select repeat('aaaaaaaaa', 20)::int;
select repeat('aaaaaaaaa', 20)::int;
select repeat('aaaaaaaaa', 20)::int;
show pg_logging.buffer_position;
select * from logging.get_log(true);

select logging.test_ereport('error', 'one', 'two', 'three');
select * from logging.get_log(true);

drop extension pg_logging;
