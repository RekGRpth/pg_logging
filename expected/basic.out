create schema logging;
create extension pg_logging schema logging;
set pg_logging.buffer_position = 0;
ERROR:  this parameter could not be changed
set pg_logging.buffer_size = 2000;
ERROR:  buffer size cannot be increased
set pg_logging.buffer_size = 1;
show pg_logging.buffer_position;
 pg_logging.buffer_position 
----------------------------
 0
(1 row)

select logging.flush_log();
 flush_log 
-----------
 
(1 row)

select * from logging.pg_log;
 level | errno | message | detail | hint 
-------+-------+---------+--------+------
       |       |         |        | 
(1 row)

select 1/0;
ERROR:  division by zero
show pg_logging.buffer_position;
 pg_logging.buffer_position 
----------------------------
 44
(1 row)

select 'aaaaa'::int;
ERROR:  invalid input syntax for integer: "aaaaa"
LINE 1: select 'aaaaa'::int;
               ^
show pg_logging.buffer_position;
 pg_logging.buffer_position 
----------------------------
 112
(1 row)

select * from logging.get_log(true);
 level | errno |                  message                  | detail | hint | position 
-------+-------+-------------------------------------------+--------+------+----------
    20 |     0 | division by zero                          |        |      |        0
    20 |     0 | invalid input syntax for integer: "aaaaa" |        |      |       44
(2 rows)

select repeat('aaaaaaaaa', 20)::int;
ERROR:  invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
select repeat('aaaaaaaaa', 20)::int;
ERROR:  invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
select repeat('aaaaaaaaa', 20)::int;
ERROR:  invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
select repeat('aaaaaaaaa', 20)::int;
ERROR:  invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
show pg_logging.buffer_position;
 pg_logging.buffer_position 
----------------------------
 64
(1 row)

select * from logging.get_log(true);
 level | errno |                                                                                                         message                                                                                                          | detail | hint | position 
-------+-------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------+----------
    20 |     0 | invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" |        |      |      112
    20 |     0 | invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" |        |      |      356
    20 |     0 | invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" |        |      |      600
    20 |     0 | invalid input syntax for integer: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" |        |      |      844
(4 rows)

